state/factory/workingset.go

```diff
@@ -183,17 +183,16 @@ func (ws *workingSet) Revert(snapshot int) error {
 
 // Commit persists all changes in RunActions() into the DB
 func (ws *workingSet) Commit(ctx context.Context) error {
-	err := ws.commitFunc(ws.height)
-	if !ws.Dirty() {
+	if err := ws.commitFunc(ws.height); err != nil {
```

Comment:
also seems in previous logic, ws.Reset is always called. but now it doesn't call ws.Reset on error